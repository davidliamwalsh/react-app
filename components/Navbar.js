import { Component } from 'react'
import { withRouter } from 'next/router'
import Link from 'next/link'

class Navbar extends Component {
  constructor (props) {
    super(props)

    this.burgerClick = this.burgerClick.bind(this)
    this.state = {
      navBarOpen: false
    }
  }

  burgerClick () {
    this.setState({ navBarOpen: !this.state.navBarOpen })
  }
  renderTransparantNavBar () {
    return this.props.router.pathname === '/' ? 'c-nav-active' : ''
  }
  
  renderBurgerMenu () {
    return this.state.navBarOpen ? 'c-nav--active' : ''
  }

  render () {
    return <nav className={`c-nav ${this.renderTransparantNavBar()}`}>
      <div className="c-nav__logo">
        <h4>
          <Link href="/">
            <a>Blogger</a>
          </Link>
        </h4>
      </div>
      <ul className={`c-nav__links c-nav__home-links ${this.renderBurgerMenu()}`}> 
        <li className="c-nav__links-list">
          <Link href='/'>
            <a>Home</a>
          </Link>
        </li>
        <li className="c-nav__links-list">
          <Link href='/about'>
            <a>About</a>
          </Link>
        </li>
        <li className="c-nav__links-list">
          <Link href='/contact'>
            <a>Contact</a>
          </Link>
        </li>
      </ul>
      <div className="c-nav__burger" onClick={this.burgerClick}>
        <div className="c-nav__burger-top"></div>
        <div className="c-nav__burger-middle"></div>
        <div className="c-nav__burger-bottom"></div>
      </div>
    </nav>
  }
}

export default withRouter(Navbar)