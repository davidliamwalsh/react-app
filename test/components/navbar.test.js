import Navbar from '../../components/Navbar'

describe('<Navbar />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <Navbar />
    )

    expect(component).toMatchSnapshot()
  })
}) 