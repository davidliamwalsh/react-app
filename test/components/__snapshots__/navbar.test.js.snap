// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> it renders correctly 1`] = `ShallowWrapper {}`;